syntax = "proto3";

package proto_serialization;

message TransportCatalogue{
    repeated Bus buses = 1;
    repeated Stop stops = 2;
    repeated Distance distances = 3;
}

message Bus {
    string name = 1;
    bool is_roundtrip = 2;
    repeated int32 stop_ids = 3;
}

message Stop {
    int32 id = 1;
    string name = 2;
    double latitude = 3;
    double longitude = 4;
}

message Distance {
    int32 stop_from_id = 1;
    int32 stop_to_id = 2;
    double distance = 3;
}